### DISCUSSION TITLE
[Bug] Pagination for dependent packages limited to 11 pages

### DISCUSSION THREAD

**Discussion by 7rulnik** [community/community#128448] [npm] [answered]
For now, it's impossible to get all dependents for packages because pagination limited to 11 pages.

Let's take [source-map](https://www.npmjs.com/package/source-map) for example. We can see 2371 dependents on the package page.
![image](https://user-images.githubusercontent.com/5969049/106433815-bda1fa00-6481-11eb-8296-ca4fe70a9836.png)

If we will follow [dependend page](https://www.npmjs.com/browse/depended/source-map) we can see pagination.
In devtool we can see that npm uses XHR to fetch new page. So we can fetch all pages in programmatic way:
```js
async function getAllDependents(packageName) {
    const packages = []
    let offset = 0
    let json

    do {
        const response = await fetch(`https://www.npmjs.com/browse/depended/${packageName}?offset=${offset}`, {
            "headers": {
                "x-spiferack": "1"
            },
        });

        json = await response.json()
        offset += json.paginationSize
        packages.push(...json.packages)
    } while (json.hasNext)

    return packages
}
```

Let's run in it:
```
f = await getAllDependents('source-map')
```

But we got: 396 items (11 pages × 36 item)
![image](https://user-images.githubusercontent.com/5969049/106434240-55074d00-6482-11eb-9f9f-1217904f8a72.png)

If we will try to set offset by hand we will have `pagination limit reached` error:
https://www.npmjs.com/browse/depended/source-map?offset=396
![image](https://user-images.githubusercontent.com/5969049/106436847-a49b4800-6485-11eb-8ee2-922befb10fd1.png)

**Comment by MylesBorins** [comment, marked as answer]
hey @7rulnik thanks for opening this issue, I'm bringing it up with the registry / website engineering team.

**Reply by MylesBorins** [reply]
@7rulnik fixing this is not something we are prioritizing right now but we do have it tracked on our engineering backlog
